src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContractSerializer.cs

```diff
@@ -33,13 +33,27 @@ public sealed class DataContractSerializer : XmlObjectSerializer
         private static SerializationOption s_option = IsReflectionBackupAllowed() ? SerializationOption.ReflectionAsBackup : SerializationOption.CodeGenOnly;
         private static bool s_optionAlreadySet;
 
-        internal static UTF8Encoding UTF8NoBom { get; } = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: false);
-        internal static UTF8Encoding ValidatingUTF8 { get; } = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
+        internal sealed class SealedUTF8Encoding : UTF8Encoding
+        {
+            public SealedUTF8Encoding(bool encoderShouldEmitUTF8Identifier, bool throwOnInvalidBytes)
+                : base(encoderShouldEmitUTF8Identifier, throwOnInvalidBytes)
+            { }
+        }
+
+        internal sealed class SealedUnicodeEncoding : UnicodeEncoding
+        {
+            public SealedUnicodeEncoding(bool bigEndian, bool byteOrderMark, bool throwOnInvalidBytes)
+                : base(bigEndian, byteOrderMark, throwOnInvalidBytes)
+            { }
+        }
+
+        internal static SealedUTF8Encoding UTF8NoBom { get; } = new SealedUTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: false);
+        internal static SealedUTF8Encoding ValidatingUTF8 { get; } = new SealedUTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
```

Comment:
This was originally a temporary part of moving vector code to Encoding class.

This change seems to give unexpected improvement to xml serialisation for `ClassImplementingIXmlSerialiable` so (it is also used in XmlConverter) so I left it in the code.